src/Analyzers/Core/Analyzers/UseCollectionInitializer/AbstractObjectCreationExpressionAnalyzer.cs

```diff
@@ -167,7 +167,8 @@ protected bool ExpressionContainsValuePatternOrReferencesInitializedSymbol(Synta
         {
             foreach (var subExpression in expression.DescendantNodesAndSelf().OfType<TExpressionSyntax>())
             {
-                if (!_syntaxFacts.IsNameOfMemberAccessExpression(subExpression))
```

Comment:
this was badly named in two ways:

1. it counted for MemberBindingExpressions, not just MemberAccessExpressions
2. it wouldn't return true for all MemberAccessExpressions, only those whose kind is SimpleMemberAccessExprssion.

As such, this method was removed entirely.  It was replaced with IsNameOfSimpleMemberAccessExpression (which now does excactly what it says), and IsNameOfMemberBindingExpression.  Code that called the prior method now calls both of these methods to preserve semantics.

Note: this means we should nto have a change in behavior here.  However, it does not mean these features are correct.  It's highly possible they're inappropriately handling a node they shouldn't be.